embassy-rp/src/multicore.rs

```diff
@@ -255,23 +264,41 @@ where
 
     // Wait until the other core has copied `entry` before returning.
     fifo_read();
+
+    // Enable FIFO interrupts from core1.
+    #[cfg(feature = "rp2040")]
+    unsafe {
+        interrupt::SIO_IRQ_PROC0.enable()
+    };
+    #[cfg(feature = "_rp235x")]
+    unsafe {
+        interrupt::SIO_IRQ_FIFO.enable()
+    };
 }
 
 /// Pause execution on CORE1.
 pub fn pause_core1() {
     if IS_CORE1_INIT.load(Ordering::Acquire) {
+        cortex_m::interrupt::disable();
         fifo_write(PAUSE_TOKEN);
         // Wait for CORE1 to signal it has paused execution.
         while fifo_read() != PAUSE_TOKEN {}
+        unsafe {
```

Comment:
might this lose a `SCHEDULE_TOKEN`?

Reply:
Good catch!
Added logic to hand the token to the user handler if it's not the `PAUSE_TOKEN`: 8f09c54

Reply:
Nice!